include/openPMD/backend/Attribute.hpp

```diff
@@ -150,6 +235,12 @@ getCast( Attribute const & a )
 {
     auto v = a.getResource();
 
+    // icpc does not like variantSrc::visit
+    // we use variantSrc::visit for the other compilers to avoid having an
+    // endless list of if-then-else
+    // also, once we switch to C++17, we might throw this out in
+    // favor of a hopefully working std::visit
+#ifdef __INTEL_COMPILER
```

Comment:
The problem is only with the legacy `icpc` compiler from intel but not the new LLVM-based `icpx` compilers.
Thus, I would guard this like this:
```suggestion
#if defined(__ICC) || defined(__INTEL_COMPILER)
```
(That's what CMake uses as identifier: https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Compiler/Intel-DetermineCompiler.cmake#L2)

The IntelLLVM (`icpx`) identifier on the other hand is:
```
(defined(__clang__) && defined(__INTEL_CLANG_COMPILER)) || defined(__INTEL_LLVM_COMPILER)
```
https://gitlab.kitware.com/cmake/cmake/-/blob/v3.21.1/Modules/Compiler/IntelLLVM-DetermineCompiler.cmake#L2